### PR TITLE
added "allow" functionality to subnet pool definition

### PR DESCRIPTION
**Pull request by bp8017**
I wanted the ability to be able to limit certain subnets to a defined class (i.e. match only mac addresses of virtualbox VMs)

### CODE REVIEW

**Comment by jmccann** [comment]
Is this working for you?  I think you need to additionally update the following for this to really work:
* https://github.com/chef-brigade/dhcp-cookbook/blob/master/templates/default/subnet_pool.erb
* https://github.com/chef-brigade/dhcp-cookbook/blob/master/resources/pool.rb

And expand existing testing:
* https://github.com/chef-brigade/dhcp-cookbook/blob/master/test/cookbooks/testing/recipes/dhcp_subnet.rb
* https://github.com/chef-brigade/dhcp-cookbook/blob/master/spec/providers/dhcp_subnet_spec.rb

Thanks!

**Comment by bp8017** [comment]
Thanks for the reminder. I assumed since "deny" was in there that there would be similar functionality for "allow", but failed to check. New commits added. Please review and let me know.

File: templates/default/subnet_pool.erb [hunk 1]
```diff
@@ -9,4 +9,7 @@ pool {
   <% if @deny -%>
   deny <%= @deny %>;
   <% end -%>
+  <% if @allow -%>
+  deny <%= @allow %>;
```
Comment (jmccann) [inline templates/default/subnet_pool.erb]: You have `deny` here but I think you want `allow`

File: metadata.rb [hunk 2]
```diff
@@ -6,7 +6,7 @@
 license 'Apache 2.0'
 description 'DHCP'
 long_description IO.read(File.join(File.dirname(__FILE__), 'README.md'))
-version '5.1.0'
+version '5.1.2'
```
Comment (jmccann) [inline metadata.rb]: Please undo the change of adding a version bump

**Comment by jmccann** [comment]
Sorry it took a while for me to get back to this.  I just reviewed the code and made some inline comments.  Once those are addressed I'll probably merge it, add a little more testing, then release it.

Thanks!

**Comment by bp8017** [comment]
Updates made and committed. Let me know if you have any other questions or concerns. I couldn't seem to get the testing to work correctly, but I am using these updates successfully on a few of my servers now.

**Comment by jmccann** [comment]
Yea, don't worry about the CI testing.  I fixed it in master.  I'm testing locally for this.

You still forgot to undo the version bump in `metadata.rb`

Thanks!

**Comment by bp8017** [comment]
Done.

**Comment by jmccann** [comment]
Sweet, thanks!
